File: mobile/lib/models/reading.dart
class Reading {
  final int id;
  final int deviceId;
  final double value;
  final String unit;
  final DateTime timestamp;

  Reading({
    required this.id,
    required this.deviceId,
    required this.value,
    required this.unit,
    required this.timestamp,
  });

  factory Reading.fromJson(Map<String, dynamic> json) => Reading(
        id: json['id'] as int,
        deviceId: json['deviceId'] as int,
        value: (json['value'] as num).toDouble(),
        unit: json['unit'] as String,
        timestamp: DateTime.parse(json['timestamp'] as String),
      );
}

File: backend/Models/Reading.cs
namespace EnergyTrack.API.Models;

public class Reading
{
    public int Id { get; set; }
    public int DeviceId { get; set; }
    public double Value { get; set; }
    public string Unit { get; set; } = string.Empty;
    public DateTime Timestamp { get; set; }

    public Device Device { get; set; } = null!;
}
